### PR TITLE
Working terraform workflow with docs

### LINKED ISSUE #1: terraform templates
- need a terraform template for easy provisioning of VM resources that come installed with the folding@home agent

Need: a storage account and batch account

Terraform templates for:
resource group: https://www.terraform.io/docs/providers/azurerm/d/resource_group.html
batch account: https://www.terraform.io/docs/providers/azurerm/d/batch_account.html
storage account: https://www.terraform.io/docs/providers/azurerm/d/storage_account.html

### PR DESCRIPTION
**Pull request by mtarng**
Closes #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,9 @@
-# folding-on-azure
\ No newline at end of file
+# folding-on-azure
+
+The purpose of this repository is to provide an easy set of steps to deploy [Folding@Home](https://foldingathome.org/) resources on Azure.
```
Comment (bnookala) [inline README.md]: maybe mention Azure Batch here

File: terraform-templates/variables.tf [hunk 2]
```diff
@@ -4,42 +4,81 @@ variable "batch_account_name" {
...
   type = string
-  default = "test-folding-pool-display-name"
+  default = "azure-folding-at-home"
 }
 
 variable "batch_pool_node_agent_sku_id" {
   type = string
   default = "batch.node.ubuntu 16.04"
-  # default = "batch.node.ubuntu 18-04 LTS"
+  # default = "batch.node.ubuntu 18-04 LTS" // Targeted for GPU/graphics tasks?
```
Comment (bnookala) [inline terraform-templates/variables.tf]: can't use 18-04?

File: terraform-templates/variables.tf [hunk 3]
```diff
@@ -4,42 +4,81 @@ variable "batch_account_name" {
...
 variable "storage_account_name" {
   type = string
 }
 
+variable "fah_enable_gpu" {
+  type = bool
+  default = false
+}
+
+variable "fah_enable_smp" {
```
Comment (bnookala) [inline terraform-templates/variables.tf]: what's this flag?

Comment (mtarng) [reply terraform-templates/variables.tf]: // Enabling the SMP Core: https://foldingathome.org/2008/06/15/what-does-the-smp-core-do/

File: terraform-templates/variables.tf (hunk 2 above)
Comment (mtarng) [reply terraform-templates/variables.tf]: updated

**Pull request merged**